File: back/EGAR.Domain/Models/Order.cs
using EGAR.Domain.Enums;

namespace EGAR.Domain.Models;

public class Order
{
    public int Id { get; set; }
    public string OrderNumber { get; set; } = default!;
    public string Description { get; set; } = default!;
    public OrderStatus Status { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset UpdatedAt { get; set; }

}

File: front/src/types/order.ts
import { OrderStatus } from "./orderStatus";

export interface Order {
  id: number;
  orderNumber: string;
  description: string;
  status: OrderStatus;
  createdAt: string;
  updatedAt: string;
}

export interface CreateOrderData {
  orderNumber: string;
  description: string;
}

export interface OrderStatusChangedEvent {
  OrderId: number;
  OldStatus: OrderStatus;
  NewStatus: OrderStatus;
  ChangedAt: string;
}